File: template.ts
const day = 1

import * as dotenv from 'dotenv'
import axios from 'axios'

dotenv.config()
let session = process.env.AOC_SESSION

let fetch_input = async (day: number) : Promise<string> => {

    let url: string = `https://adventofcode.com/2022/day/${day}/input`
    let headers = { 'Cookie': `session=${session}` }
    try {
        return await axios.get(url, { headers })
    } catch (err) { throw err }
}

fetch_input(day).then( (infile) => {

    console.log(infile)
    console.log('part 1:')
    console.log('part 2:')

}).catch ((err) => { throw err })

File: template.py
import requests
import sys
import os
 
day = 1
session = os.getenv('AOC_SESSION')
url = "https://adventofcode.com/2022/day/" + str(day) + "/input"
headers = {'Cookie': 'session=' + session}
resp = requests.get(url, headers = headers)
if resp.status_code != 200:
    sys.exit(f'{resp.status_code}\n{resp.reason}\n{resp.content}')
infile = resp.text
lines = infile.split('\n')
lines.pop()
print(lines, 'len:', len(lines))

